packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,460 @@
+        return (children.length / 2).floor();
+      case PivotType.pivotBySize:
+        int pivotIndex = -1;
+        double maxSize = double.negativeInfinity;
+        for (int i = 0; i < children.length; i++) {
+          if (children[i].byteSize > maxSize) {
+            maxSize = children[i].byteSize.toDouble();
+            pivotIndex = i;
+          }
+        }
+        return pivotIndex;
+      default:
+        return -1;
+    }
+  }
+
+  /// Implements the ordered treemap algorithm studied in [this research paper](https://www.cs.umd.edu/~ben/papers/Shneiderman2001Ordered.pdf).
+  ///
+  /// **Algorithm**
+  ///
+  /// Divides a given list of treemap nodes into four parts:
+  /// L1, P, L2, L3.
+  ///
+  /// P (pivot) is the treemap node chosen to be the pivot based on the pivot type.
+  /// L1 includes all treemap nodes before the pivot treemap node.
+  /// L2 and L3 combined include all treemap nodes after the pivot treemap node.
+  /// A combination of elements are put into L2 and L3 so that
+  /// the aspect ratio of the pivot cell (P) is as close to 1 as it can be.
+  ///
+  /// **Layout**
+  /// ```
+  /// ----------------------
+  /// |      |  P   |      |
+  /// |      |      |      |
+  /// |  L1  |------|  L3  |
+  /// |      |  L2  |      |
+  /// |      |      |      |
+  /// ----------------------
+  /// ```
+  List<Positioned> buildTreemaps({
+    @required List<TreemapNode> children,
+    @required double width,
+    @required double height,
+  }) {
+    final isHorizontalRectangle = width > height;
+
+    final totalByteSize = computeByteSizeForNodes(nodes: children);
+    if (children.isEmpty) {
+      return [];
+    }
+
+    // Sort the list of treemap nodes, descending in size.
+    children.sort((a, b) => b.byteSize.compareTo(a.byteSize));
+    if (children.length <= 2) {
+      final positionedChildren = <Positioned>[];
+      double offset = 0;
+
+      for (final child in children) {
+        final ratio = child.byteSize / totalByteSize;
+
+        positionedChildren.add(
+          Positioned(
+            left: isHorizontalRectangle ? offset : 0.0,
+            top: isHorizontalRectangle ? 0.0 : offset,
+            width: isHorizontalRectangle ? ratio * width : width,
+            height: isHorizontalRectangle ? height : ratio * height,
+            child: NewTreemap.fromRoot(
+              rootNode: child,
+              levelsVisible: widget.levelsVisible - 1,
+              onRootChangedCallback: widget.onRootChangedCallback,
+            ),
+          ),
+        );
+        offset += isHorizontalRectangle ? ratio * width : ratio * height;
+      }
+
+      return positionedChildren;
+    }
+
+    final pivotIndex = computePivot(children);
+
+    final pivotDataReference = children[pivotIndex];
+    final pByteSize = pivotDataReference.byteSize;
+
+    final list1 = children.sublist(0, pivotIndex);
+    final list1ByteSize = computeByteSizeForNodes(nodes: list1);
+
+    List list2 = <TreemapNode>[];
```

Comment:
use var instead of List. The type will be inferred by the RHS. here and below for list3